### PR TITLE
[macos] add reties to CodeQL download step

### PR DESCRIPTION
**Pull request by ilia-shipitsin**
# Description

add reties to CodeQL download step

## Check list
- [ ] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/macos/provision/core/codeql-bundle.sh [hunk 1]
```diff
@@ -2,8 +2,8 @@
 source ~/utils/utils.sh
 
 # Retrieve the CLI version of the latest CodeQL bundle.
-base_url="$(curl -fsSL https://raw.githubusercontent.com/github/codeql-action/v2/src/defaults.json)"
-bundle_version="$(echo "$base_url" | jq -r '.cliVersion')"
+download_with_retries https://raw.githubusercontent.com/github/codeql-action/v2/src/defaults.json "/tmp" "codeql-defaults.json"
+bundle_version="$(cat /tmp/codeql-defaults.json | jq -r '.cliVersion')"
```
Comment (henrymercer) [inline images/macos/provision/core/codeql-bundle.sh:6]: We can simplify this:
```suggestion
bundle_version="$(jq -r '.cliVersion' /tmp/codeql-defaults.json)"
```

File: images/macos/provision/core/codeql-bundle.sh [hunk 2]
```diff
@@ -2,8 +2,8 @@
 source ~/utils/utils.sh
 
 # Retrieve the CLI version of the latest CodeQL bundle.
-base_url="$(curl -fsSL https://raw.githubusercontent.com/github/codeql-action/v2/src/defaults.json)"
-bundle_version="$(echo "$base_url" | jq -r '.cliVersion')"
+download_with_retries https://raw.githubusercontent.com/github/codeql-action/v2/src/defaults.json "/tmp" "codeql-defaults.json"
```
Comment (henrymercer) [inline images/macos/provision/core/codeql-bundle.sh:5]: Could you confirm that `download_with_retries` follows redirects?

**Review by henrymercer — APPROVED** [review]
Thanks!

File: images/macos/provision/core/codeql-bundle.sh (hunk 2 above)
Comment (ilia-shipitsin) [reply images/macos/provision/core/codeql-bundle.sh:5]: no idea on that. it uses curl under the hood. hopefully the same behaviour as regular curl

do we have tests on codeql (triggered on image generation) ?

Comment (ilia-shipitsin) [reply images/macos/provision/core/codeql-bundle.sh:5]: I hope that if something will go wrong with redirect, we'll catch it with https://github.com/actions/runner-images/blob/main/images/macos/provision/core/codeql-bundle.sh#L28

Comment (henrymercer) [reply images/macos/provision/core/codeql-bundle.sh:5]: Looks like it does — https://github.com/actions/runner-images/blob/e113e2f6b829f0e6fd528b93a1241c52410873a2/images/macos/provision/utils/utils.sh#L15C39-L15C40 passes `-L`.

The runner image tests will check the download works, I just wanted to be robust to the case where the download doesn't include a redirect today but starts including a redirect in the future.

**Review by henrymercer — APPROVED** [review]

**Review by Alexey-Ayupov — APPROVED** [review]

**Pull request merged**